### PR TITLE
Add backend validations

### LINKED ISSUE #25: Enable ajv validation
## Context

In frontend we use [`rjsf`](https://www.npmjs.com/package/@rjsf/core) that validate json with [`ajv`](https://github.com/ajv-validator/ajv).

We must:

- [x] Enable `ajv` in backend before sending email
- [x] Enable `validation` with `rjsf`

### PR DESCRIPTION
**Pull request by Its-Alex**
Closes #25 

~~What's missing:~~

~~- [x] Tests~~

What is done in this pull request:

- Add backend validation
- Enable frontend validation since they were bypassed

### CODE REVIEW

**Comment by Its-Alex** [comment]
@stephane-klein Need review

File: backend/src/app.js [hunk 1]
```diff
@@ -197,20 +198,32 @@ router.post('/v1/templates/:name/send/:stmpSelected?', async (ctx) => {
...
         mjmlTemplatePath = path.join(templatePath, 'default.mjml');
         txtTemplatePath = path.join(templatePath, 'default.txt');
     }
 
-    const templates = [subjectTemplatePath, mjmlTemplatePath, txtTemplatePath];
+    const templates = [
+        subjectTemplatePath,
+        mjmlTemplatePath,
+        txtTemplatePath,
+        jsonSchemaPath
```
Comment (stephane-klein) [inline backend/src/app.js:225]: @Its-Alex **suggestion:** what do you think about make `jsonSchemaPath` file optional? If `jsonSchemaPath` file missing, then schema validation is skipped with a `console.info` messages 🤔

Comment (stephane-klein) [reply backend/src/app.js:225]: @Its-Alex Bad idea, this file is mandatory for frontend formulaire rendering.

**Comment by stephane-klein** [comment]
@Its-Alex I have this error:

```
$ direnv allow
$ cd backend
$ yarn run test
yarn run v1.22.10
$ eslint --ignore-path .gitignore .
$ jest
  ●  process.exit called with "1"

      315 |                 console.error(error);
      316 |                 console.error('Smtp url:', getSmtpUrl());
    > 317 |                 process.exit(1);
          |                         ^
      318 |             } else {
      319 |                 console.log('Smtp server is ready to take our messages');
      320 |             }

      at src/app.js:317:25
      at SMTPConnection.<anonymous> (node_modules/nodemailer/lib/smtp-transport/index.js:346:24)
      at SMTPConnection._onError (node_modules/nodemailer/lib/smtp-connection/index.js:764:14)
      at Socket.SMTPConnection._onSocketError (node_modules/nodemailer/lib/smtp-connection/index.js:190:45)

 RUNS  test/routes.test.js
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Its-Alex** [comment]
> @Its-Alex **issue:** I have this error:
> 
> ```
> $ direnv allow
> $ cd backend
> $ yarn run test
> yarn run v1.22.10
> $ eslint --ignore-path .gitignore .
> $ jest
>   ●  process.exit called with "1"
> 
>       315 |                 console.error(error);
>       316 |                 console.error('Smtp url:', getSmtpUrl());
>     > 317 |                 process.exit(1);
>           |                         ^
>       318 |             } else {
>       319 |                 console.log('Smtp server is ready to take our messages');
>       320 |             }
> 
>       at src/app.js:317:25
>       at SMTPConnection.<anonymous> (node_modules/nodemailer/lib/smtp-transport/index.js:346:24)
>       at SMTPConnection._onError (node_modules/nodemailer/lib/smtp-connection/index.js:764:14)
>       at Socket.SMTPConnection._onSocketError (node_modules/nodemailer/lib/smtp-connection/index.js:190:45)
> 
>  RUNS  test/routes.test.js
> error Command failed with exit code 1.
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> ```

@stephane-klein You must start mailcatcher instance to run tests, this is a weird bug that has always been here

**Comment by stephane-klein** [comment]
> You must start at least one mailcatcher instance to run tests, this is a weird bug that has always been here

Fixed by https://github.com/spacefill/gibbon-mail/commit/12e9f314bddb00ba59a810a14d145b460c756717

**Review by stephane-klein — APPROVED** [review]
Reviewed, thanks @Its-Alex 👍

**Pull request merged**